formats/json/commonMain/src/kotlinx/serialization/json/internal/FormatLanguage.kt

```diff
@@ -0,0 +1,45 @@
+package kotlinx.serialization.json.internal;
+
+import kotlinx.serialization.InternalSerializationApi
+
+/**
+ * Multiplatform analogue of `org.intellij.lang.annotations.Language` annotation.
+ *
+ * An alias is used instead of classes, because the actual class in the JVM will overload the classes from the stdlib -
+ * and different classes with the same fully-qualified name can be loaded in different loaders.
+ * [see](https://github.com/JetBrains/java-annotations/issues/34)
+ *
+ * Specifies that an element of the program represents a string that is a source code on a specified language.
+ * Code editors may use this annotation to enable syntax highlighting, code completion and other features
+ * inside the literals that assigned to the annotated variables, passed as arguments to the annotated parameters,
+ * or returned from the annotated methods.
+ * <p>
+ * This annotation also could be used as a meta-annotation, to define derived annotations for convenience.
+ * E.g. the following annotation could be defined to annotate the strings that represent Java methods:
+ *
+ * <pre>
+ *   &#64;Language(value = "JAVA", prefix = "class X{", suffix = "}")
+ *   &#64;interface JavaMethod {}
+ * </pre>
+ * <p>
+ * Note that using the derived annotation as meta-annotation is not supported.
+ * Meta-annotation works only one level deep.
+ */
+
+@InternalSerializationApi
+@Retention(AnnotationRetention.BINARY)
+@Target(
+    AnnotationTarget.FUNCTION,
+    AnnotationTarget.PROPERTY_GETTER,
+    AnnotationTarget.PROPERTY_SETTER,
+    AnnotationTarget.FIELD,
+    AnnotationTarget.VALUE_PARAMETER,
+    AnnotationTarget.LOCAL_VARIABLE,
+    AnnotationTarget.ANNOTATION_CLASS
+)
+public expect annotation class FormatLanguage(
```

Comment:
I think `@OptionalExpectation` may help here with avoiding re-declaring annotation on JS and Native

Reply:
Why? I think it's available for any annotation

Reply:
Because `org.jetbrains:annotations` is a transitive dependency from JVM stdlib.

Reply:
Because in this case, I'm not sure that JSON syntax highlighting will work correctly for non-JVM projects, because it works specifically for annotations with the name `org.intellij.lang.annotations.Language` but this is `kotlinx.serialization.json.internal.FormatLanguage`.